zerver/views/realm_export.py

```diff
@@ -97,20 +95,17 @@ def get_realm_exports(request: HttpRequest, user: UserProfile) -> HttpResponse:
 @require_realm_admin
 def delete_realm_export(request: HttpRequest, user: UserProfile, export_id: int) -> HttpResponse:
     try:
-        audit_log_entry = RealmAuditLog.objects.get(
-            id=export_id, realm=user.realm, event_type=AuditLogEventType.REALM_EXPORTED
-        )
-    except RealmAuditLog.DoesNotExist:
+        export_row = RealmExport.objects.get(id=export_id)
```

Comment:
Is it possible that a client makes an API request to delete `export_id` (which refers to id of `RealmAuditLog`) and at the same instant migration completes and the request now tries to delete an object with incorrect `id` (of `RealmExport`) ?

Do we discard all the API requests for the period of migration & django restart during deployment ?

Understanding the exact order in which things happen during deployment would be helpful here, I think!

Reply:
It's very unlikely that a relevant API request of this form happens during the migration, just given their frequency. But generally migrations are applied before the server restarts, rather than the other way around.